ClosedXML/Excel/Cells/XLCell.cs

```diff
@@ -362,92 +361,144 @@ public string GetFormattedString()
+                    if (referenceCell.IsEmpty(false))
+                        return 0;
                     else
-                    {
-                        sName = Worksheet.Name;
-                        cAddress = fA1;
-                    }
+                        return referenceCell.Value;
+                }
 
-                    object retVal;
-                    try
-                    {
-                        IsEvaluating = true;
+                retVal = Worksheet.Evaluate(fA1);
 
-                        if (_worksheet
-                                .Workbook
-                                .WorksheetsInternal
-                                .Any<XLWorksheet>(w => String.Compare(w.Name, sName, true) == 0)
-                            && XLHelper.IsValidA1Address(cAddress))
-                        {
-                            var referenceCell = _worksheet.Workbook.Worksheet(sName).Cell(cAddress);
-                            if (referenceCell.IsEmpty(false))
-                                return 0;
-                            else
-                                return referenceCell.Value;
-                        }
+            }
+            finally
+            {
+                IsEvaluating = false;
+            }
 
-                        retVal = Worksheet.Evaluate(fA1);
-                    }
-                    finally
-                    {
-                        IsEvaluating = false;
-                    }
+            var retValEnumerable = retVal as IEnumerable;
+            if (retValEnumerable != null && !(retVal is String))
+                return retValEnumerable.Cast<object>().First();
 
-                    var retValEnumerable = retVal as IEnumerable;
+            return retVal;
+        }
 
-                    if (retValEnumerable != null && !(retVal is String))
-                        return retValEnumerable.Cast<object>().First();
+        public void InvalidateFormula()
+        {
+            NeedsRecalculation = true;
+            Worksheet.Workbook.InvalidateFormulas();
+            ModifiedAtVersion = Worksheet.Workbook.RecalculationCounter;
+        }
 
-                    return retVal;
-                }
+        /// <summary>
+        /// Perform an evaluation of cell formula. If cell does not contain formula nothing happens, if cell does not need
+        /// recalculation (<see cref="NeedsRecalculation"/> is False) nothing happens either, unless <paramref name="force"/> flag is specified.
+        /// Otherwise recalculation is perfomed, result value is preserved in <see cref="CachedValue"/> and returned.
+        /// </summary>
+        /// <param name="force">Flag indicating whether a recalculation must be performed even is cell does not need it.</param>
+        /// <returns>Null if cell does not contain a formula. Calculated value otherwise.</returns>
+        public object Evaluate(bool force = false)
+        {
+            if (NeedsRecalculation || force)
```

Comment:
`if (force || NeedsRecalculation)` be better to avoid `NeedsRecalculation` to be evaluated when `force == true`.